pkgs/top-level/all-packages.nix

```diff
@@ -25438,12 +25438,10 @@ in
 
   kontemplate = callPackage ../applications/networking/cluster/kontemplate { };
 
+  # Only keep the last three minor versions around that correspond to the last three supported kubernetes versions:
+  # https://kubernetes.io/docs/setup/release/version-skew-policy/#supported-versions
   inherit (callPackage ../applications/networking/cluster/kops {})
     mkKops
```

Comment:
If you rely on outdated kubernetes cluster versions, there is still a `mkKops` function, which should be easy to use.

Reply:
Agreed in principle, with one exception:
Every once in a while, there will be an update that will be more complex than just bumping the k8s version, eg. the etcd 2 -> 3 migration. For these, the typical advice is to not skip updating to that specific version because it's the only version where you can be reasonably sure it's well tested (eg. jumping from 1.11 to 1.16 is probably not tested by anyone).
So we should make sure that depending on what the upgrade instructions say, to keep the least minimum version around to ensure that upgrading your Nix(OS) version is safe in that sense.

Reply:
I changed the wording to mention upgrades.